pkg/profiler/cpu/maps.go

```diff
@@ -167,37 +181,46 @@ func (m *bpfMaps) setUnwindTable(pid int, ut unwind.UnwindTable) error {
 			return fmt.Errorf("write the program counter: %w", err)
 		}
 
+		// Write __reserved_do_not_use.
+		if err := binary.Write(buf, m.byteOrder, uint16(0)); err != nil {
+			return fmt.Errorf("write CFA register bytes: %w", err)
+		}
+
 		// Write CFA.
 		switch row.CFA.Rule {
 		case frame.RuleCFA:
 			// Write CFA register.
-			if err := binary.Write(buf, m.byteOrder, row.CFA.Reg); err != nil {
+			var CFARegister uint16
+			if row.CFA.Reg == 6 {
```

Comment:
nit: Let's use some constants instead of magic numbers here.

Reply:
Good shout Was planning to do it but it seems I forgot about it